### ISSUE TITLE
Running on iOS 

### ISSUE THREAD

**Issue by bjnortier** [usefulsensors/openai-whisper#4]
@nyadla-sys Moving the conversation here from the C++ implementation.

I've made some progress, I have a prototype working on iOS with different bits hacked together. But the CoreML delegate doesn't work. The Interpreter() doesn't initialise when created with the delegate. I'm trying to figure out why.

**Comment by nyadla-sys** [comment]
have you tried GPU delegate, and also minimum TF version must be greater than >2.4

**Comment by nyadla-sys** [comment]
However, I've made some effort toward creating a full Int8 model, and its size is approximately 38MB. I'll release my code soon.

**Comment by bjnortier** [comment]
Yes I've tried GPU delegate but it doesn't load your model. I think it's because the output is int32:

"The Core ML delegate currently supports float (FP32 and FP16) models."

**Comment by nyadla-sys** [comment]
Thanks @bjnortier for implementing basic iOS app using whisper.tflite 

source code for iOS APP development (whisper.tflite hybrid model, 40MB model size)
https://github.com/bjnortier/whisper-tflite-ios

**Comment by bjnortier** [comment]
It’s a very rough prototype but it works

**Comment by nyadla-sys** [comment]
https://apps.apple.com/in/app/whisper-asr/id6444556326

Pls download iOS app using whisper tflite model

**Comment by NickDarvey** [comment]
@nyadla-sys, is [whisper-asr](https://apps.apple.com/in/app/whisper-asr/id6444556326) based on the [bjnortier/whisper-tflite-ios](https://github.com/bjnortier/whisper-tflite-ios) repo?

**Comment by nyadla-sys** [comment]
Yes and used part of the code

**Comment by nyadla-sys** [comment]
Whisper ASR iOS app is available now and here is the link 
https://apps.apple.com/in/app/whisper-asr/id6444556326

**Comment by ankushg** [comment]
> Yes I've tried GPU delegate but it doesn't load your model. I think it's because the output is int32:
> 
> "The Core ML delegate currently supports float (FP32 and FP16) models."

Has there been any progress on creating a model that works with the CoreML Delegate?

**Comment by nyadla-sys** [comment]
@ankushg Please use this [notebook](https://colab.research.google.com/drive/1CElRlLNbe9dRYpmu58jkbtu5lUKhMPRS?usp=sharing) to generate fp32 model that is compatible for coreml delegate

**Comment by jj09** [comment]
@bjnortier thanks for sample code for iOS app! Is there a way to correlate translated text with timestamps?

**Comment by bjnortier** [comment]
@jj09 not in the TFLite version and it processes the whole 30 second chunk in one go. Whisper.cpp has better timestamp support.

**Comment by jj09** [comment]
@bjnortier are there any samples how to do it?

**Comment by bjnortier** [comment]
Whisper.cpp contains ObjC and SwiftUI examples
